src/components/Game.tsx

```diff
@@ -0,0 +1,158 @@
+import { useEffect, useReducer, useState } from 'react';
+import { calculateWinner, createArrayAttributes } from '../utilities/';
+import Board from './Board';
+import GameInfo from './GameInfo';
+import Move from './Move';
+import Footer from './Footer';
+
+/**
+ * Type for game state properties
+ */
+type GameState = {
+  history: GameHistory;
+  xIsNext: boolean;
+  stepNumber: number;
+  counter: number;
+};
+
+/**
+ * Initilization of game state properties
+ * @returns
+ */
+const initialStateProperties: GameState = {
+  history: createArrayAttributes(9),
+  xIsNext: true,
+  stepNumber: 0,
+  counter: 0,
+};
+
+/**
+ * Handle game logic
+ * @param state
+ * @param action
+ * @returns
+ */
+const reducer = (state: GameState, action: GameStateAction): GameState => {
```

Comment:
I would put this and the 2 types above either below this component, or throw this reducer into it's own file. This file is for the Game component, so the Game component should be at the top